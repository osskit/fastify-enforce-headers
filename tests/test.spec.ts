import Fastify from 'fastify';
import { enforceHeaders } from '../src/index.js';

describe('main', () => {
  it('should enforce missing default headers', async () => {
    const fastify = Fastify();

    await fastify.register(enforceHeaders);

    fastify.get('/', async () => ({
      bla: 'hey',
    }));

    const res = await fastify.inject({ method: 'GET', path: '/' });

    expect(res.statusCode).toMatchSnapshot();
    expect(res.json()).toMatchSnapshot();
  });

  it('should allow default headers', async () => {
    const fastify = Fastify();
    fastify.get('/', async () => ({
      bla: 'hey',
    }));

    await fastify.register(enforceHeaders);

    const res = await fastify.inject({ method: 'GET', path: '/', headers: { 'x-api-client': 'bla', 'x-api-client-version': '1.0' } });

    expect(res.statusCode).toMatchSnapshot();
    expect(res.json()).toMatchSnapshot();
  });

  it('should enforce missing custom headers', async () => {
    const fastify = Fastify();
    fastify.get('/', async () => ({
      bla: 'hey',
    }));

    await fastify.register(enforceHeaders, { headers: ['x-custom-header'] });

    const res = await fastify.inject({ method: 'GET', path: '/', headers: { 'x-api-client': 'bla', 'x-api-client-version': '1.0' } });

    expect(res.statusCode).toMatchSnapshot();
    expect(res.json()).toMatchSnapshot();
  });

  it('should allow custom headers', async () => {
    const fastify = Fastify();
    fastify.get('/', async () => ({
      bla: 'hey',
    }));

    await fastify.register(enforceHeaders, { headers: ['x-custom-header'] });

    const res = await fastify.inject({ method: 'GET', path: '/', headers: { 'x-custom-header': 'bla' } });

    expect(res.statusCode).toMatchSnapshot();
    expect(res.json()).toMatchSnapshot();
  });
});
