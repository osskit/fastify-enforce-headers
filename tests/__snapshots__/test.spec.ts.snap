// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`main > should allow custom headers 1`] = `200`;

exports[`main > should allow custom headers 2`] = `
{
  "bla": "hey",
}
`;

exports[`main > should allow default headers 1`] = `200`;

exports[`main > should allow default headers 2`] = `
{
  "bla": "hey",
}
`;

exports[`main > should enforce missing custom headers 1`] = `400`;

exports[`main > should enforce missing custom headers 2`] = `
{
  "error": "Bad Request",
  "message": "Missing required headers in request",
  "statusCode": 400,
}
`;

exports[`main > should enforce missing default headers 1`] = `400`;

exports[`main > should enforce missing default headers 2`] = `
{
  "error": "Bad Request",
  "message": "Missing required headers in request",
  "statusCode": 400,
}
`;
